diff --git a/src/rp2_common/lwip/CMakeLists.txt b/src/rp2_common/lwip/CMakeLists.txt
index 33a4ca6..d584108 100644
--- a/src/rp2_common/lwip/CMakeLists.txt
+++ b/src/rp2_common/lwip/CMakeLists.txt
@@ -56,4 +56,8 @@ if (EXISTS ${LWIP_PATH}/${LWIP_TEST_PATH})
         ${LWIP_PATH}/src/include/ipv4
         ${LWIP_PATH}/src/include/lwip/apps
     )
+
+    target_link_libraries(lwip INTERFACE
+        pico_stdlib
+    )
 endif()
