diff --git a/src/core/dns.c b/src/core/dns.c
index f83ea9b4..768f72ea 100644
--- a/src/core/dns.c
+++ b/src/core/dns.c
@@ -100,7 +100,7 @@
 /** Random generator function to create random TXIDs and source ports for queries */
 #ifndef DNS_RAND_TXID
 #if ((LWIP_DNS_SECURE & LWIP_DNS_SECURE_RAND_XID) != 0)
-#define DNS_RAND_TXID LWIP_RAND
+#define DNS_RAND_TXID LWIP_RAND_WIZ
 #else
 static u16_t dns_txid;
 #define DNS_RAND_TXID() (++dns_txid)
